src/test/test.sh

```diff
@@ -0,0 +1,10 @@
+g++ -std=c++17 -Wall -I catch.hpp -I ../ -I ../../vendor/realm-core/src/realm/object-store/ \
```

Comment:
Please add some comments about what this script does.  It's not obvious to me at least.
Also, should it be moved to the main `scripts/test.sh` as an parameter?